api/python/provisioner/commands/validator/validator.py

```diff
@@ -255,3 +259,72 @@ class YumRepoDataValidator(DirValidator):
     def __init__(self):
         super().__init__({Path("repomd.xml"): FileValidator(required=True)},
                          required=True)
+
+
+@attr.s
+class HashSumValidator(PathValidator):
+
+    """
+    Validator of hash-sum for the provided file and expected hash-sum for this
+    file.
+
+    Attributes
+    ----------
+    hash_sum: Union[str, bytes, bytearray]
+        Hexadecimal string or byte-array object with expected hash-sum value
+        of validated file.
+    hash_type: HashType
+        Type of hash sum. See `Hashtype` for more information
+
+    """
+    _hash_sum: Union[str, bytes, bytearray] = attr.ib(
+        validator=attr.validators.instance_of((str, bytes, bytearray))
+    )
+    _hash_type: HashType = attr.ib(
+        validator=attr.validators.in_(HashType),
+        default=HashType.MD5,
+        converter=lambda x: HashType.MD5 if x is None else HashType(x)
```

Comment:
Yes, we can pass `None` here in case if `hash_type` was not specified in the file or string. Consider leaving such a converter. It allows us not to validate in code the `hash_type is None` condition to not pass this parameter to the validator.